libnavigation-core/api/current.txt

```diff
@@ -54,8 +54,10 @@ package com.mapbox.navigation.core {
   }
 
   public final class MapboxNavigationProvider {
-    method public static com.mapbox.navigation.core.MapboxNavigation create(com.mapbox.navigation.base.options.NavigationOptions navigationOptions);
-    method public static com.mapbox.navigation.core.MapboxNavigation retrieve();
+    method @kotlin.jvm.Synchronized public static com.mapbox.navigation.core.MapboxNavigation create(com.mapbox.navigation.base.options.NavigationOptions navigationOptions);
+    method @kotlin.jvm.Synchronized public static void destroy();
+    method @kotlin.jvm.Synchronized public static boolean isCreated();
+    method @kotlin.jvm.Synchronized public static com.mapbox.navigation.core.MapboxNavigation retrieve();
```

Comment:
can we remove synchronized for now and add the `isCreated()` and `destroy()` functions?

we should separate the conversation for thread safety. since it is a `Singleton`, we should still be in the discussion phase, rather than committing to api definitions. Consider that common use cases are already unsafe from multiple threads.

``` kotlin
if (!MapboxNavigationProvider.isCreated()) {
    MapboxNavigationProvider.create(...)
}
```

If i was a client evaluating mapbox, i'd immediately complain about that. And we're looking to make this interface a required interface.